test/test_tornado.py

```diff
@@ -50,27 +50,19 @@ def _get_response(self, url: str, benchmark: Callable) -> Response:
             status=r.code, headers=list(r.headers.get_all()), content=r.body
         )
 
-    def assert_404(self, url: str):
-        with pytest.raises(HTTPClientError) as exc:
+    def assert_error_code(self, url: str, error_code: int):
+        with pytest.raises(HTTPClientError) as e:
             self._get_response(url, lambda f: f())
-        r = exc.value.response
+        r = e.value.response
         assert r is not None
-
-        res = Response(status=r.code, headers=list(r.headers.get_all()), content=r.body)
-        assert res.status == 404
-        content = decode_response(res)
-        assert isinstance(content, dict) and isinstance(content["message"], str)
-
-    def assert_403(self, url: str):
-        with pytest.raises(HTTPClientError) as exc:
-            self._get_response(url, lambda f: f())
-        r = exc.value.response
-        assert r is not None
-
-        res = Response(status=r.code, headers=list(r.headers.get_all()), content=r.body)
-        assert res.status == 403
-        content = decode_response(res)
-        assert isinstance(content, dict) and isinstance(content["message"], str)
+        assert_error_response(
```

Comment:
The assertions are now done in `assert_error_response` so that the only job of `assert_error_code` is to reconstruct the `Response` from the raised error